src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1381,6 +1381,20 @@
   (let [last-modified-milliseconds (.lastModified (io/file cacrl))]
        (time-coerce/from-long last-modified-milliseconds)))
 
+(schema/defn ^:always-validate update-crl :- CertificateRevocationList
+  [upstream-crl-pem :- InputStream
+   crl-path :- schema/Str]
+  (log/info (i18n/trs "Updating CRL at {0}" crl-path))
+  (let [upstream-crls (utils/pem->crls upstream-crl-pem)
+        our-crl (first (utils/pem->crls crl-path))
```

Comment:
`our-crl` here will already contain the upstream ones. We need to remove those and swap in the new ones.

Reply:
Oh I missed the `first`, I'm sorry.